server/ssl.c

```diff
@@ -123,6 +205,13 @@ AP_DECLARE(void) ap_setup_ssl_optional_fns(apr_pool_t *pool)
     module_ssl_var_lookup = (fn_ssl_var_lookup
         && fn_ssl_var_lookup != ssl_var_lookup)? fn_ssl_var_lookup : NULL;
     APR_REGISTER_OPTIONAL_FN(ssl_var_lookup);
+
+    module_ssl_proxy_enable = APR_RETRIEVE_OPTIONAL_FN(ssl_proxy_enable);
+    APR_REGISTER_OPTIONAL_FN(ssl_proxy_enable);
```

Comment:
Doesn't APR_RETRIEVE_OPTIONAL_FN() always return NULL if it's called before APR_REGISTER_OPTIONAL_FN()?
I'm likely missing something here :p

Reply:
We are in `ap_setup_ssl_optional_fns`, which is called by `core_post_config` which is the first runner of the post config hook. Modules register optional functions typically in register hooks which runs before post config. Hence the pointers should be set if someone did.